docs/src/stories/components/ToggleSwitch/ToggleSwitch.stories.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+
+export default {
+  title: 'Components/ToggleSwitch',
+  parameters: {
+    layout: 'padded'
+  },
+  excludeStories: ['ToggleSwitchTemplate'],
+  controls: { expanded: true },
+  argTypes: {
+    checked: {
+      control: {type: 'boolean'},
+      description: 'checkbox state'
+    },
+    disabled: {
+      description: 'disabled field',
+      control: {type: 'boolean'}
+    },
+    size: {
+      options: ['medium', 'small'],
+      control: {
+        type: 'inline-radio'
+      },
+      description: 'size'
+    },
+    labelPosition: {
+      options: ['start', 'end'],
+      control: {
+        type: 'inline-radio'
+      },
+      description: 'label position'
+    }
+  }
+}
+
+function classNamesForSwitch(disabled, checked, size, labelPosition) {
+  const classNames = ['ToggleSwitch'];
+
+  if (checked) {
+    classNames.push("ToggleSwitch--checked")
+  }
+  if (disabled) {
+    classNames.push("ToggleSwitch--disabled")
+  }
+  if (size === 'small') {
+    classNames.push("ToggleSwitch--small")
+  }
+  if (labelPosition === 'end') {
+    classNames.push('flex-row-reverse')
+  } else {
+    classNames.push('flex-row')
+  }
+
+  return classNames.join(' ')
+}
+
+export const ToggleSwitchTemplate = ({disabled, checked, size, labelPosition}) => (
+  <>
+    <toggle-switch class={classNamesForSwitch(disabled, checked, size, labelPosition)}>
+      <span aria-hidden="true" className="ToggleSwitch-labels">
+        <div className="text-right ToggleSwitch-onLabel" style={{visibility: checked ? 'visible' : 'hidden' }}>On</div>
+        <div className="text-right ToggleSwitch-offLabel" style={{visibility: checked ? 'hidden' : 'visible' }}>Off</div>
+      </span>
+
+      <button
+        className="ToggleSwitch-track"
+        role="switch"
+        aria-checked={checked ? 'true' : 'false'}
+        aria-disabled={disabled ? "true" : "false"}>
+        <span class="sr-only">{checked ? "On" : "Off"}</span>
```

Comment:
Instead of using `sr-only` to visually hide this text, should we use `aria-label` instead? And we might wanna describe it a bit more like `aria-label="Toggle is on"` to make it more clear.

Reply:
I think `aria-label` works, but we should check with somebody from the Accessibility team about `"Toggle is on"` vs `"On"`.

`role="switch"` might cause screen readers to say something that will sound redundant.

cc @ichelsea

Reply:
Having `role="switch"` here is great! It takes the checkbox properties from checked/unchecked to On/Off with no intermediate state. We shouldn't need `aria-label` or screen-reader only text, that would be redundant. One of the cool things that adding `role="switch"` and `aria-checked` means that the state of the button is already announced to the user, no extra code needed 🙂

Reply:
I'm not sure. I'm copying the React implementation here, which uses `<VisuallyHidden>`. The equivalent in PVC/PCSS is `sr-only`. @mperrotti do you know what the right thing to do here is?

Reply:
Never mind, looks like @ichelsea already knows what's up!